src/main/java/uk/gov/hmcts/cft/idam/testingsupportapi/service/TestingCaseWorkerProfileService.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.hmcts.cft.idam.testingsupportapi.service;
+
+import org.springframework.jms.core.JmsTemplate;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.cft.idam.api.v2.common.model.User;
+import uk.gov.hmcts.cft.idam.testingsupportapi.repo.TestingEntityRepo;
+import uk.gov.hmcts.cft.idam.testingsupportapi.repo.model.TestingEntityType;
+import uk.gov.hmcts.cft.rd.api.RefDataCaseWorkerApi;
+import uk.gov.hmcts.cft.rd.model.CaseWorkerLocation;
+import uk.gov.hmcts.cft.rd.model.CaseWorkerProfile;
+import uk.gov.hmcts.cft.rd.model.CaseWorkerRole;
+import uk.gov.hmcts.cft.rd.model.CaseWorkerService;
+import uk.gov.hmcts.cft.rd.model.CreateCaseWorkerProfileRequest;
+
+import java.util.Collections;
+import java.util.List;
+
+@Service
+public class TestingCaseWorkerProfileService extends TestingEntityService<CaseWorkerProfile> {
+
+    private final RefDataCaseWorkerApi refDataCaseWorkerApi;
+
+    public TestingCaseWorkerProfileService(TestingEntityRepo testingEntityRepo,
+                                           JmsTemplate jmsTemplate,
+                                           RefDataCaseWorkerApi refDataCaseWorkerApi) {
+        super(testingEntityRepo, jmsTemplate);
+        this.refDataCaseWorkerApi = refDataCaseWorkerApi;
+    }
+
+    public User createCaseWorkerProfile(String sessionId, User user) {
+        CreateCaseWorkerProfileRequest createRequest = convertToCreateRequest(user);
+        refDataCaseWorkerApi.createCaseWorkerProfile(createRequest);
+
+        createTestingEntity(sessionId, createRequest);
+
+        return user;
+    }
+
+    private CreateCaseWorkerProfileRequest convertToCreateRequest(User user) {
+        CreateCaseWorkerProfileRequest createRequest = new CreateCaseWorkerProfileRequest();
+        createRequest.setCaseWorkerId(user.getId());
+        createRequest.setFirstName(user.getForename());
+        createRequest.setLastName(user.getSurname());
+        createRequest.setEmail(user.getEmail());
+
+        createRequest.setSkills(Collections.emptyList());
+        createRequest.setUserType("CTSC");
+        createRequest.setUserProfileIdamStatus("ACTIVE");
+        createRequest.setStaffAdmin(user.getRoleNames().contains("staff-admin"));
```

Comment:
Would we ever want testing-support-api clients to send us things like roles, user types, regions, locations?

Reply:
Probably yes, but we don't have a "V2" model for that (yet). I think it's fine to leave these as defaults for now since IDAM is the only user. When we want to expand the usage of this we can discuss a newer input model